### ISSUE TITLE
Classification Results

### ISSUE THREAD

**Issue by sheshap** [liruihui/PointAugment#8]
I downloaded the code and trained the pointnet model from scratch as per the instructions in the README file.

However, I could only see **Best Accuracy: 0.890600**

Please help us reproduce the results mentioned in the paper.

Thanks in Advance.

**Comment by liruihui** [comment]
Due to adversarial training, it is not very stable. There is also a large improvement space on loss design and architecture.
You may try again and it can be even better than the papers' result. 
If you find some error, pls let me know.

**Comment by YU-YUE-HUA** [comment]
> 我按照README文件中的说明下载了代码并从头开始训练Pointnet模型。
> 
> 但是，我只能看到**最高精度：0.890600**
> 
> 请帮助我们重现本文中提到的结果。
> 
> 提前致谢。

你好打扰了 我想知道在
  clsLoss = loss_utils.cls_loss(pred_pc, pred_aug, target, pc_tran, aug_tran, pc_feat,
                                              aug_feat, ispn=ispn)
                clsLoss.backward(retain_graph=True)
                optimizer_c.step()
这一部分 您在复现的实现有没有出问题

**Comment by sheshap** [comment]
@YU-YUE-HUA in English Please

**Comment by YU-YUE-HUA** [comment]
> @ YU-YUE-HUA英文请

Hello, excuse me. I'd like to know
ClsLoss = LOSs_utils.CLs_loss (PREd_PC, Pred_AUG, target, PC_TRAN, aug_TRAN, PC_FEAT,
Aug_feat, ISPN = ISPN)
Clsloss.backward (retain_graph = true)
Optimizer_c. Step ()
Do you have any problems with this part of your replicating implementation？

**Comment by sheshap** [comment]
I didn't face any problems. I have re-created PointAugment+PointNet results 90.3%

**Comment by YU-YUE-HUA** [comment]
Hello, I wonder if you can use your code for reference. Thank you very much.

**Comment by sheshap** [comment]
if you download the code and train it (maybe more than once from scratch) you will be able to re-create PointAugment+PointNet results.

If you want to re-create PointAugment+PointNet++ results then download the forked version from my repository and train from scratch. https://github.com/sheshap/PointAugment.git

I just got 92.9498% for PointAugment & PointNet++

**Comment by YU-YUE-HUA** [comment]
> 如果您下载代码并对其进行培训（可能从头开始，可能不止一次），您将能够重新创建PointAugment + PointNet结果。
> 
> 如果要重新创建PointAugment + PointNet ++结果，请从我的存储库中下载分叉的版本，然后从头开始进行培训。https://github.com/sheshap/PointAugment.git
> 
> 我刚获得PointAugment和PointNet ++的92.9498％

When I reproduced the Point Augment code, I gave an error in the loss function of classifier and intensifier, which led to the failure of the code reproduction. So I want to refer to your code, I wonder if it is OK?Thank you in advance

**Comment by sheshap** [comment]
@YU-YUE-HUA sure. Please use it.

**Comment by YU-YUE-HUA** [comment]
> @ YU-YUE-HUA当然。请使用它。

I encountered this error during training:”one of the variables needed for gradient computation has been modified by an inplace operation: [torch.FloatTensor [256, 4]], which is output 0 of TBackward, is at version 2; expected version 1 instead. Hint: the backtrace further above shows the operation that failed to compute its gradient. The variable in question was changed in there or anywhere later. Good luck!“I've tried many times, but I still can't solve it，can you help me？Thank you very much！！！

**Comment by Kartik17** [comment]
@YU-YUE-HUA I encountered the same issue, I install the anaconda environment as per the readme and the issue went away. I guess it has to do with the version of PyTorch.
